setup.py

```diff
@@ -12,28 +12,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from setuptools import setup
+from setuptools import setup, find_namespace_packages
 
-with open("README.md", "r") as fh:
+with open("README.md", "rt", encoding="utf8") as fh:
     long_description = fh.read()
 
+# ops.__init__ needs to pull in ops.charm to work around a circular
+# import so we can't import it here as that pulls in yaml which isn't
+# necessarily there yet.
+import ast
+version = 'unknown'
```

Comment:
Please:
- import `ast` at the top
- encapsulate all this code in a function
- provide a test case in `test_infra.py` for this (one way to easily do it is call `setup.py` to print the version), so we're sure it works in all the Pythons we have in Travis